modules/core-components/src/test/java/org/mule/runtime/core/privileged/processor/chain/DefaultMessageProcessorChainTestCase.java

```diff
@@ -0,0 +1,304 @@
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.slf4j.Logger;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+@SmallTest
+@SuppressWarnings("deprecation")
+public class DefaultMessageProcessorChainTestCase extends AbstractReactiveProcessorTestCase {
+
+  private static final Logger LOGGER = getLogger(DefaultMessageProcessorChainTestCase.class);
+
+  protected MuleContext muleContext;
+
+  private final AtomicInteger nonBlockingProcessorsExecuted = new AtomicInteger(0);
+  private final ProcessingStrategyFactory processingStrategyFactory;
+
+  @Rule
+  public ExpectedException expectedException = none();
+
+  @Parameterized.Parameters(name = "{0}, {2}")
+  public static Collection<Object[]> parameters() {
+    return asList(new Object[][] {
+        {"TransactionAwareStreamEmitterProcessingStrategyFactory",
+            new TransactionAwareStreamEmitterProcessingStrategyFactory(), BLOCKING},
+        {"TransactionAwareProactorStreamEmitterProcessingStrategyFactory",
+            new TransactionAwareProactorStreamEmitterProcessingStrategyFactory(), BLOCKING},
+        {"StreamEmitterProcessingStrategyFactory",
+            new StreamEmitterProcessingStrategyFactory(), BLOCKING},
+        {"ProactorStreamEmitterProcessingStrategyFactory",
+            new ProactorStreamEmitterProcessingStrategyFactory(), BLOCKING},
+        {"BlockingProcessingStrategyFactory",
+            new BlockingProcessingStrategyFactory(), BLOCKING},
+        {"DirectProcessingStrategyFactory",
+            new DirectProcessingStrategyFactory(), BLOCKING},
+        {"TransactionAwareStreamEmitterProcessingStrategyFactory",
+            new TransactionAwareStreamEmitterProcessingStrategyFactory(), NON_BLOCKING},
+        {"TransactionAwareProactorStreamEmitterProcessingStrategyFactory",
+            new TransactionAwareProactorStreamEmitterProcessingStrategyFactory(), NON_BLOCKING},
+        {"StreamEmitterProcessingStrategyFactory",
+            new StreamEmitterProcessingStrategyFactory(), NON_BLOCKING},
+        {"ProactorStreamEmitterProcessingStrategyFactory",
+            new ProactorStreamEmitterProcessingStrategyFactory(), NON_BLOCKING},
+        {"BlockingProcessingStrategyFactory",
+            new BlockingProcessingStrategyFactory(), NON_BLOCKING},
+        {"DirectProcessingStrategyFactory",
+            new DirectProcessingStrategyFactory(), NON_BLOCKING}});
+  }
+
+  private Flow flow;
+
+  public DefaultMessageProcessorChainTestCase(String psName, ProcessingStrategyFactory processingStrategyFactory, Mode mode) {
+    super(mode);
+    this.processingStrategyFactory = processingStrategyFactory;
+  }
+
+  @Before
+  public void before() throws MuleException {
+    nonBlockingProcessorsExecuted.set(0);
+    muleContext = spy(super.muleContext);
```

Comment:
this is an static reference, access through the class -> AbstractMuleContextTestCase.muleContext.